HDK v0.0.101 won't compile on MacOS. libsqlite3-sys can't compile for wasm32-unknown-unknown

I've been working on an HDK wrapping rust crate, but I now realize it won't compile for the wasm target while building on MacOS, because of `libsqlite3-sys` sneaking in as a dependency, via `hdk_derive -> holochain_zome_types -> holo_hash -> rusqlite -> libsqlite3-sys`

Can be seen in CI here: https://github.com/lightningrodlabs/hdk_crud/runs/3257245529#step:4:252

From the `main` branch of hdk_crud (https://github.com/lightningrodlabs/hdk_crud) run the following to reproduce.

`cargo build --release --target wasm32-unknown-unknown`

```
   Compiling proc-macro2 v1.0.28
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.74
   Compiling autocfg v1.0.1
   Compiling version_check v0.9.3
   Compiling libc v0.2.98
   Compiling serde v1.0.123
   Compiling serde_derive v1.0.123
   Compiling autocfg v0.1.7
   Compiling cfg-if v1.0.0
   Compiling ryu v1.0.5
   Compiling getrandom v0.1.16
   Compiling memchr v2.4.0
   Compiling serde_json v1.0.64
   Compiling bitflags v1.2.1
   Compiling rand_core v0.4.2
   Compiling pkg-config v0.3.19
   Compiling cc v1.0.69
   Compiling once_cell v1.8.0
   Compiling lazy_static v1.4.0
   Compiling smallvec v1.6.1
   Compiling unicode-segmentation v1.8.0
   Compiling cfg-if v0.1.10
   Compiling fnv v1.0.7
   Compiling byteorder v1.4.3
   Compiling strsim v0.9.3
   Compiling ident_case v1.0.1
   Compiling scopeguard v1.1.0
   Compiling itoa v0.4.7
   Compiling paste v1.0.5
   Compiling ppv-lite86 v0.2.10
   Compiling convert_case v0.4.0
   Compiling regex-syntax v0.6.25
   Compiling derive_builder v0.9.0
   Compiling either v1.6.1
   Compiling log v0.4.14
   Compiling constant_time_eq v0.1.5
   Compiling arrayvec v0.5.2
   Compiling strum v0.18.0
   Compiling fallible-iterator v0.2.0
   Compiling fallible-streaming-iterator v0.1.9
   Compiling arrayref v0.3.6
   Compiling pin-project-lite v0.1.12
   Compiling base64 v0.13.0
   Compiling predicates-core v1.0.2
   Compiling subtle v2.4.1
   Compiling normalize-line-endings v0.3.0
   Compiling difference v2.0.0
   Compiling treeline v0.1.0
   Compiling ansi_term v0.12.1
   Compiling downcast v0.10.0
   Compiling fragile v1.0.0
   Compiling instant v0.1.10
   Compiling tracing-core v0.1.18
   Compiling sharded-slab v0.1.1
   Compiling rand_chacha v0.1.1
   Compiling rand_pcg v0.1.2
   Compiling rand v0.6.5
   Compiling ahash v0.7.4
   Compiling value-bag v1.0.0-alpha.7
   Compiling num-traits v0.2.14
   Compiling indexmap v1.7.0
   Compiling num-integer v0.1.44
   Compiling rand_core v0.3.1
   Compiling rand_os v0.1.3
   Compiling rand_jitter v0.1.4
   Compiling lock_api v0.3.4
   Compiling lock_api v0.4.4
   Compiling parking_lot_core v0.7.2
   Compiling heck v0.3.3
   Compiling itertools v0.8.2
   Compiling blake2b_simd v0.5.11
   Compiling thread_local v1.1.3
   Compiling predicates-tree v1.0.2
   Compiling libsqlite3-sys v0.22.2
   Compiling parking_lot_core v0.8.3
   Compiling rand_isaac v0.1.1
   Compiling rand_hc v0.1.0
   Compiling rand_xorshift v0.1.1
   Compiling parking_lot v0.10.2
   Compiling time v0.1.43
   Compiling rand_core v0.5.1
   Compiling regex-automata v0.1.10
   Compiling parking_lot v0.11.1
   Compiling aho-corasick v0.7.18
   Compiling quote v1.0.9
   Compiling getrandom v0.2.3
   Compiling rand_chacha v0.2.2
   Compiling matchers v0.0.1
The following warnings were emitted during compilation:

warning: sqlite3/sqlite3.c:14230:10: fatal error: 'stdio.h' file not found
warning: #include <stdio.h>
warning:          ^~~~~~~~~
warning: 1 error generated.

error: failed to run custom build command for `libsqlite3-sys v0.22.2`

Caused by:
  process didn't exit successfully: `/Users/x/x/libs/hdk_crud/target/release/build/libsqlite3-sys-577f092052a96113/build-script-build` (exit code: 1)
  --- stdout
  cargo:rerun-if-changed=sqlite3/sqlite3.c
  cargo:rerun-if-changed=sqlite3/wasm32-wasi-vfs.c
  cargo:rerun-if-env-changed=SQLITE_MAX_VARIABLE_NUMBER
  cargo:rerun-if-env-changed=SQLITE_MAX_EXPR_DEPTH
  cargo:rerun-if-env-changed=LIBSQLITE3_FLAGS
  TARGET = Some("wasm32-unknown-unknown")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-apple-darwin")
  CC_wasm32-unknown-unknown = None
  CC_wasm32_unknown_unknown = None
  TARGET_CC = None
  CC = None
  CFLAGS_wasm32-unknown-unknown = None
  CFLAGS_wasm32_unknown_unknown = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  running: "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=wasm32-unknown-unknown" "-DSQLITE_CORE" "-DSQLITE_DEFAULT_FOREIGN_KEYS=1" "-DSQLITE_ENABLE_API_ARMOR" "-DSQLITE_ENABLE_COLUMN_METADATA" "-DSQLITE_ENABLE_DBSTAT_VTAB" "-DSQLITE_ENABLE_FTS3" "-DSQLITE_ENABLE_FTS3_PARENTHESIS" "-DSQLITE_ENABLE_FTS5" "-DSQLITE_ENABLE_JSON1" "-DSQLITE_ENABLE_LOAD_EXTENSION=1" "-DSQLITE_ENABLE_MEMORY_MANAGEMENT" "-DSQLITE_ENABLE_RTREE" "-DSQLITE_ENABLE_STAT2" "-DSQLITE_ENABLE_STAT4" "-DSQLITE_SOUNDEX" "-DSQLITE_THREADSAFE=1" "-DSQLITE_USE_URI" "-DHAVE_USLEEP=1" "-D_POSIX_THREAD_SAFE_FUNCTIONS" "-DHAVE_ISNAN" "-DHAVE_LOCALTIME_R" "-o" "/Users/x/x/libs/hdk_crud/target/wasm32-unknown-unknown/release/build/libsqlite3-sys-e21ba9ed8d63f021/out/sqlite3/sqlite3.o" "-c" "sqlite3/sqlite3.c"
  cargo:warning=sqlite3/sqlite3.c:14230:10: fatal error: 'stdio.h' file not found
  cargo:warning=#include <stdio.h>
  cargo:warning=         ^~~~~~~~~
  cargo:warning=1 error generated.
  exit code: 1

  --- stderr


  error occurred: Command "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=wasm32-unknown-unknown" "-DSQLITE_CORE" "-DSQLITE_DEFAULT_FOREIGN_KEYS=1" "-DSQLITE_ENABLE_API_ARMOR" "-DSQLITE_ENABLE_COLUMN_METADATA" "-DSQLITE_ENABLE_DBSTAT_VTAB" "-DSQLITE_ENABLE_FTS3" "-DSQLITE_ENABLE_FTS3_PARENTHESIS" "-DSQLITE_ENABLE_FTS5" "-DSQLITE_ENABLE_JSON1" "-DSQLITE_ENABLE_LOAD_EXTENSION=1" "-DSQLITE_ENABLE_MEMORY_MANAGEMENT" "-DSQLITE_ENABLE_RTREE" "-DSQLITE_ENABLE_STAT2" "-DSQLITE_ENABLE_STAT4" "-DSQLITE_SOUNDEX" "-DSQLITE_THREADSAFE=1" "-DSQLITE_USE_URI" "-DHAVE_USLEEP=1" "-D_POSIX_THREAD_SAFE_FUNCTIONS" "-DHAVE_ISNAN" "-DHAVE_LOCALTIME_R" "-o" "/Users/x/x/libs/hdk_crud/target/wasm32-unknown-unknown/release/build/libsqlite3-sys-e21ba9ed8d63f021/out/sqlite3/sqlite3.o" "-c" "sqlite3/sqlite3.c" with args "clang" did not execute successfully (status code exit code: 1).


warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/crates/hdk_derive/Cargo.toml
+++ b/crates/hdk_derive/Cargo.toml
@@ -18,7 +18,7 @@ syn = { version = "1", features = [ "full", "extra-traits" ] }
 quote = "1"
 proc-macro2 = "1"
 paste = "=1.0.5"
-holochain_zome_types = { version = "0.0.4", path = "../holochain_zome_types" }
+holochain_zome_types = { version = "0.0.4", path = "../holochain_zome_types", default-features = false }
 
 [features]
 default = []

